cobc/parser.y

```diff
@@ -13949,33 +14014,33 @@ replacing_item:
 
 rep_keyword:
   /* empty */
-| ALL				{ inspect_keyword = 1; }
-| LEADING			{ inspect_keyword = 2; }
-| FIRST				{ inspect_keyword = 3; }
-| TRAILING			{ inspect_keyword = 4; }
+| ALL				{ inspect_keyword = INSPECT_REP_ALL; }
+| LEADING			{ inspect_keyword = INSPECT_REP_LEADING; }
+| FIRST				{ inspect_keyword = INSPECT_REP_FIRST; }
+| TRAILING			{ inspect_keyword = INSPECT_REP_TRAILING; }
 ;
 
 replacing_region:
   inspect_from BY inspect_to inspect_region
   {
 	switch (inspect_keyword) {
-		case 1:
-			$$ = cb_build_replacing_all ($1, $3, $4);
-			break;
-		case 2:
-			$$ = cb_build_replacing_leading ($1, $3, $4);
-			break;
-		case 3:
-			$$ = cb_build_replacing_first ($1, $3, $4);
-			break;
-		case 4:
-			$$ = cb_build_replacing_trailing ($1, $3, $4);
-			break;
-		default:
-			cb_error_x (CB_TREE (current_statement),
-				    _("INSPECT missing ALL/FIRST/LEADING/TRAILING"));
-			$$ = cb_build_replacing_all ($1, $3, $4);
-			break;
+	case INSPECT_REP_ALL:
```

Comment:
I actually tried to do this as it indeed would be cleaner. However I did not find a way to avoid having to give one type declaration for every non-terminal symbol; bison's documentation seems to explicitly require explicit declarations as soon as more than one type is involved for semantic values.
Unless I'm missing something, I feel such a change would better fit in its own PR.

Reply:
Agreed with the own PR, I think the default is the first entry of the union, all non-defaults need a type (or explicit cast as in `$<enum cb_inspect_type>0`, which _may_ also need the union definition containing the type).